libbeat/publisher/queue/diskqueue/serialize_test.go

```diff
@@ -53,17 +89,29 @@ func TestSerialize(t *testing.T) {
 
 		// Use decoder to decode the serialized bytes.
 		decoder := newEventDecoder()
+		decoder.serializationFormat = test.format
 		buf := decoder.Buffer(len(serialized))
 		copy(buf, serialized)
 		decoded, err := decoder.Decode()
 		if err != nil {
 			t.Fatalf("[%v] Couldn't decode serialized data: %v", test.name, err)
 		}
 
-		decodedValue, err := decoded.Content.Fields.GetValue("test_field")
-		if err != nil {
-			t.Fatalf("[%v] Couldn't get field 'test_field': %v", test.name, err)
+		switch test.format {
+		case SerializationCBOR:
+			event, ok := decoded.(publisher.Event)
+			assert.True(t, ok)
```

Comment:
If the type is wrong, we should interrupt the test, hence `assert` is not the best option here:
```suggestion
			event, ok := decoded.(publisher.Event)
			require.True(t, ok)
```